ostd/src/mm/page/allocator.rs

```diff
@@ -234,6 +233,74 @@ impl<const ORDER: usize> BuddyFrameAllocator<ORDER> {
 
         self.allocated -= count;
     }
+
+    /// set_frames_allocated
+    ///
+    /// # Description
+    ///
+    /// Given frames, described by a range of **frame number** [start, end),
+    /// mark them as allocated. Make sure they can be correctly deallocated
+    /// afterwards, while will not be allocated before deallocation.
+    ///
+    /// # Panics
+    ///
+    /// The function panics if no suitable block found for the given range.
+    pub fn set_frames_allocated(&mut self, start: usize, end: usize) {
```

Comment:
It is fine if the caller violates the pre-condition above since the function has already considered this. 

Essentially, the function traverses all the free lists until it finds one free block containing at least one input-described frame. If there is no suitable block, the function will panic. 

It is important to note that the free lists can only be initialized through the function `add_frame`. Therefore, all the found blocks come from the frames previously added by calling `add_frame`. You can refer to the code and the comments within for more details : )